tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -30,11 +30,18 @@ var (
 )
 
 func getS3Object(ctx context.Context, labels map[string]string) (io.ReadCloser, error) {
-	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(labels["bucket_region"]))
-	if err != nil {
-		return nil, err
+	var s3Client *s3.Client
+
+	if c, ok := s3Clients[labels["bucket_region"]]; ok {
```

Comment:
do we only need a single client per bucket region? there isn't the possibility of different buckets in the same region requiring different clients for auth reasons?

Reply:
I suspect we can ignore this for now as multiple regions/bucket auth can be attached to this lambda.

Reply:
I think multiple buckets with different auth options would increase the complexity, it would be easier to configure another lambda with another role